lucene/core/src/java/org/apache/lucene/store/Directory.java

```diff
@@ -165,4 +165,13 @@ public void copyFrom(Directory from, String src, String dest, IOContext context)
    * @throws AlreadyClosedException if this Directory is closed
    */
   protected void ensureOpen() throws AlreadyClosedException {}
+  
```

Comment:
This change was made to eliminate the "instanceof HdfsDirectory" in the fileModified() method of BlockDirectory.  The polymorphism solution seemed best, but as you say, means modifying Lucene classes (all the relevant subclasses of Directory are in Lucene, except HdfsDirectory)...